crates/bevy_reflect/src/serde/deserialize_reflect.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{FromType, Reflect, TypeRegistry};
+use serde::{Deserialize, Deserializer};
+
+/// Trait used to provide finer control when deserializing a reflected type with one of
+/// the reflection deserializers.
+///
+/// This trait is the reflection equivalent of `serde`'s [`Deserialize`] trait.
+/// The main difference is that this trait provides access to the [`TypeRegistry`],
+/// which means that we can use the registry and all its stored type information
+/// to deserialize our type.
+///
+/// This can be useful when writing a custom reflection deserializer where we may
+/// want to handle parts of the deserialization process, but temporarily pass control
+/// to the standard reflection deserializer for other parts.
+///
+/// Without this trait and its associated [type data], such a deserializer would have to
+/// write out all of the deserialization logic itself, possibly including
+/// unnecessary code duplication and trivial implementations.
```

Comment:
This is maybe for a `# Rationale` or `# Details` section. It's important to document why this type exists, for maintainers and learners, but for actually using it we don't care. So I'd separate the first part, which is the core documentation of what it is and how to use, from those two paragraph about why it came to be (design docs).

Reply:
Fair point!